src/Servers/Kestrel/Transport.MsQuic/src/Internal/MsQuicStatusHelper.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Runtime.InteropServices;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.MsQuic.Internal
+{
+    internal static class MsQuicStatusHelper
+    {
+        internal static bool Succeeded(uint status)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return status <= 0x80000000;
+            }
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return (int)status <= 0;
```

Comment:
Are negative statuses actually considered successful for the linux MsQuic library? That seems unusual.

Reply:
Yep